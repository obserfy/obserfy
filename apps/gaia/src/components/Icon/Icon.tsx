import { FC } from "react"

const Icon: FC<{
  src: string
  className?: string
  color?: string
}> = ({ src, className, color = "bg-black" }) => (
  <div
    role="img"
    aria-hidden="true"
    className={`w-5 h-5 ${color} ${className}`}
    style={{
      maskImage: `url(${src})`,
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      "-webkit-mask-image": `url(${src})`,

      maskSize: "100%",
      "-webkit-mask-size": "100%",

      maskRepeat: "no-repeat",
      "-webkit-mask-repeat": "no-repeat",
    }}
  />
)

export default Icon
