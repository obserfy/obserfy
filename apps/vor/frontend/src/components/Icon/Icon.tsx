import { FC, PropsWithoutRef } from "react"
import { Image, ImageProps } from "theme-ui"

interface Props extends PropsWithoutRef<ImageProps> {
  fill?: string
  size?: string | number
}
export const Icon: FC<Props> = ({ size = "icon", fill, sx, ...props }) => (
  <Image
    sx={{
      ...sx,
      fill: fill || "icon",
      flexShrink: 0,
      width: size,
      height: size,
    }}
    {...props}
  />
)

export default Icon
