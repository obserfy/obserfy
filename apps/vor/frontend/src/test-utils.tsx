/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
