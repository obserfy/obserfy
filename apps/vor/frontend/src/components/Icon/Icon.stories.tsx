import { FC } from "react"
import Icon from "./Icon"

export default {
  title: "Core/Icon",
  component: Icon,
  parameters: {
    componentSubtitle: "Just a simple Icon",
  },
}

export const Basic: FC = () => <Icon />
