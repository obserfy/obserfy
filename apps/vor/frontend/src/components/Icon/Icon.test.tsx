import { render } from "../../test-utils"
import { Basic } from "./Icon.stories"

describe("Icon", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
