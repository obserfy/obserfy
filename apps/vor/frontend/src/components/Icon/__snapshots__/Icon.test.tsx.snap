// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

<div>
  <div
    class="emotion-0"
  >
    <img
      class="emotion-1"
    />
  </div>
</div>
`;
